app/models/ability.rb

```diff
@@ -0,0 +1,71 @@
+class Ability
```

Comment:
Rather than comment on every piece of this, it would be better to discuss it to understand how it works.